central/imagecomponentedge/datastore/datastore_sac_test.go

```diff
@@ -0,0 +1,368 @@
+		fetchedMatches := 0
+		s.Equal(expectedFetchedSize, len(fetched))
+		for _, edge := range fetched {
+			if edge.GetId() == targetEdge1ID {
+				fetchedMatches++
+				s.Equal(expectedSrc1ID, edge.GetImageId())
+				s.Equal(expectedDst1ID, edge.GetImageComponentId())
+			}
+			if edge.GetId() == targetEdge2ID {
+				fetchedMatches++
+				s.Equal(expectedSrc2ID, edge.GetImageId())
+				s.Equal(expectedDst2ID, edge.GetImageComponentId())
+			}
+		}
+		s.Equal(expectedFetchedSize, fetchedMatches)
+	}
+}
+
+func (s *imageComponentEdgeDatastoreSACTestSuite) TestCount() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.T().Skip("graph.Context wrapper missing in ImageComponentEdge searcher",
+			"to enable Search test case in non-postgres mode")
+	}
+	err := s.dackboxTestStore.PushImageToVulnerabilitiesGraph(waitForIndexing)
+	defer s.cleanImageToVulnerabilityGraph(waitForIndexing)
+	s.Require().NoError(err)
+
+	for _, c := range testCases {
+		ctx := s.testContexts[c.contextKey]
+		expectedCount := 0
+		for _, visible := range c.expectedEdgeFound {
+			if visible {
+				expectedCount++
+			}
+		}
+		count, err := s.datastore.Count(ctx)
+		s.NoError(err)
+		s.Equal(expectedCount, count)
+	}
+}
+
+func (s *imageComponentEdgeDatastoreSACTestSuite) TestSearch() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.T().Skip("graph.Context wrapper missing in ImageComponentEdge searcher",
+			"to enable Search test case in non-postgres mode")
+	}
+	err := s.dackboxTestStore.PushImageToVulnerabilitiesGraph(waitForIndexing)
+	defer s.cleanImageToVulnerabilityGraph(waitForIndexing)
+	s.Require().NoError(err)
+
+	for _, c := range testCases {
+		ctx := s.testContexts[c.contextKey]
+		expectedCount := 0
+		for _, visible := range c.expectedEdgeFound {
+			if visible {
+				expectedCount++
+			}
+		}
+		results, err := s.datastore.Search(ctx, search.EmptyQuery())
+		s.NoError(err)
+		s.Equal(expectedCount, len(results))
+		for _, r := range results {
+			s.True(c.expectedEdgeFound[r.ID])
+		}
+	}
+}
+
+func (s *imageComponentEdgeDatastoreSACTestSuite) TestSearchEdges() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.T().Skip("graph.Context wrapper missing in ImageComponentEdge searcher",
+			"to enable Search test case in non-postgres mode")
+	}
+	err := s.dackboxTestStore.PushImageToVulnerabilitiesGraph(waitForIndexing)
+	defer s.cleanImageToVulnerabilityGraph(waitForIndexing)
+	s.Require().NoError(err)
+
+	for _, c := range testCases {
+		ctx := s.testContexts[c.contextKey]
+		expectedCount := 0
+		for _, visible := range c.expectedEdgeFound {
+			if visible {
+				expectedCount++
+			}
+		}
+		results, err := s.datastore.SearchEdges(ctx, search.EmptyQuery())
+		s.NoError(err)
+		s.Equal(expectedCount, len(results))
+		for _, r := range results {
+			s.True(c.expectedEdgeFound[r.GetId()])
+		}
+	}
+
+}
+
+func (s *imageComponentEdgeDatastoreSACTestSuite) TestSearchRawEdges() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
```

Comment:
Why do we need this test? If it's not working on PG then we will remove it once we switch to PG only mode.

Reply:
So far, the tests do work against postgres.
When it comes to the old dackbox, graph info is needed in the context, otherwise the searcher panics. An example of working searcher would be the `componentcveedge` one (https://github.com/stackrox/stackrox/blob/master/central/componentcveedge/search/searcher_impl.go#L62-L64). In the datastore at hand, The graph info is missing from the context.
I see two approaches here:
- change the existing dackbox searcher code to inject the graph in context and make the datastore tests work (was done to establish the test baseline, then reverted, see the second commit).
- disable the tests in pre-postgres mode (done with the revert of the above).